### PR TITLE
Make Harvester grant DockedCondition and remove WithVoxelUnloadBody

### PR DESCRIPTION
**Pull request by PunkPun**
Split from #20168
depends on #20174

This fixes an 8 year old hack

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs [hunk 1]
```diff
@@ -98,18 +106,37 @@ public override bool Tick(Actor self)
 			throw new InvalidOperationException("Invalid harvester dock state");
 		}
 
+		public override void Cancel(Actor self, bool keepQueue = false)
+		{
+			notifyDockables.Do(d => d.Canceled());
```
Comment (teinarss) [inline OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]: Better to use foreach(var d in notifyDockables) here and in the other places for consistency.

Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]: sure

File: OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs [hunk 2]
```diff
@@ -98,18 +106,39 @@ public override bool Tick(Actor self)
...
+			notifyDockables.Do(d => d.Docked());
+			notifyDocks.Do(d => d.Docked());
+		}
+
+		void NotifyUndocked()
+		{
+			notifyDockables.Do(d => d.Undocked());
+			if (Refinery.IsAliveAndInWorld)
+				notifyDocks.Do(d => d.Undocked());
+		}
```
Comment (teinarss) [inline OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]: And here.

Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]: fixed in #20174

**Comment by PunkPun** [comment]
superseded by #20628

**Pull request closed without merging**